### PR TITLE
4026 new homepage routing

### LINKED ISSUE #4026: Activate Homepage for root route
The tcreport was copied somewhere else and the new homepage is ready to go up!  Time to remove the root tcreport and take over the route.

Acceptance Criteria:
- [x] Add routes for the home page and Learn More pages
- [x] htaccess changes (locally and deployed)
- [x] The root route brings up the new homepage
- [x] All links on the homepage and Learn More pages wired up for proper navigation
- [x] Besides the home page, routes to other tc-report static content must continue to work:
   - [x] https://talent.canada.ca/en/talent-cloud
   - [x] https://talent.canada.ca/en/talent-cloud/report
   - [x] https://talent.canada.ca/en/terms-and-conditions
   - [x] https://talent.canada.ca/en/privacy-notice
   - [x] https://talent.canada.ca/en/communities/digital-talent
   - [x] https://talent.canada.ca/en/diversity/2000/01/01/301-overview.html
   - [x] https://talent.canada.ca/en/projects/2000/01/01/101-caveat-to-begin.html
   - [x] https://talent.canada.ca/en/hiring/2000/01/01/201-overview.html

Out of scope:
- The tcreport was copied to its new home in #4034

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #4026.

# ❗ Note
This PR should only be merged after #4096.

# :wave: Introduction
This branch routes requests for the homepage (`/`, `/en` and `/fr`) to the new homepage, instead of the tc-report homepage. It also adds a link to the homepage to the nav menu. All other routes should be unaffected.

# 🤔 Possible Todo?
Our htaccess file is configured to strip the trailing slash off urls with a permanent redirect. I originally set that up, trying to follow what I understood as a best practice of enforcing consistency - always trailing slash, or never. However, the current config it doesn't seem to work in the one case of the root APP_URL. `http://localhost:8000` is redirected to `http://localhost:8000/en/`, _with_ the trailing slash. But manually entering http://localhost:8000/en/ still redirects to a version without the trailing slash. It's not a big deal, but its annoying, and maybe not great for caching? 

Here's the part of the htaccess file which normally strips the trailing slash:
```
# Don't automatically add slash (with 301 redirect) if path matches a directory
DirectorySlash off

<IfModule mod_rewrite.c>
    RewriteEngine on

    # Strip trailing slash from all urls and redirect back to public url, unless already pointing to a public folder
    RewriteCond %{REQUEST_URI} ^/(.+)/$
    RewriteCond %{REQUEST_URI} !/public/$
    RewriteCond %{REQUEST_URI} !/dist/$
    RewriteCond %{REQUEST_URI} !/tc-report/_site/.*
    RewriteRule ^ %{ENV:APP_URL}/%1 [L,R=301]
```

# :test_tube: Testing
## :spiral_notepad: Suggested Steps
1. Refresh talentsearch (`docker-compose run --rm maintenance bash refresh_talentsearch.sh`)
2. Test that the the following urls all bring you to the new homepage (in correct language):
  - [ ] http://localhost:8000
  - [ ] http://localhost:8000/
  - [ ] http://localhost:8000/en
  - [ ] http://localhost:8000/fr
  - [ ]  http://localhost:8000/en/
  - [ ] http://localhost:8000/fr/
3. Click all the links on the homepage, and confirm they go somewhere that makes sense.
4. Test that the following links still work. They are static pages part of tc-report:
  - [ ] http://localhost:8000/en/talent-cloud
  - [ ] http://localhost:8000/en/talent-cloud/report
  - [ ] http://localhost:8000/en/terms-and-conditions
  - [ ] http://localhost:8000/en/privacy-notice
  - [ ] http://localhost:8000/en/communities/digital-talent
  - [ ] http://localhost:8000/en/diversity/2000/01/01/301-overview.html
  - [ ] http://localhost:8000/en/projects/2000/01/01/101-caveat-to-begin.html
  - [ ] http://localhost:8000/en/hiring/2000/01/01/201-overview.html
5. On any other page, the Home button in the nav menu links you to the homepage
6. Switching to french, the link in the nav menu switches to Accueil

### CODE REVIEW

**Review by esizer — COMMENTED** [review]
Awesome! ⭐ I'm ready to approve this once #4442 is merged so we can test the updated links in that.

**Review by yonikid15 — CHANGES_REQUESTED** [review]
Nice work Tristan! Looks great. I found some issue, most of which of which are out of scope.
- Styling: Nav menu is now dropping to lower level and needs a refactoring.
- Bug: There's a flash of unloaded nav menu items after signing in. We should add the Pending component to prevent that.
 (I'll drop a quick fix for the previous 2 issues, hopefully that's cool with you 😄).
 
- Bug: Homepage link to IAP page is incorrect. Directed me to 404 page.
- Bug: After reaching 404 page, user cannot use click to go back button on browser.

**Comment by esizer** [comment]
> * Bug: After reaching 404 page, user cannot use click to go back button on browser.

Yeah, this bug has been noted in https://github.com/GCTC-NTGC/gc-digital-talent/issues/4411

**Review by yonikid15 — APPROVED** [review]
Approved by me, however take a look at the commit I made before merging.

File: frontend/talentsearch/src/js/components/Router.tsx
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/Router.tsx:601]: If you hit this homepage and get a 401 (a bad token for example) you get have no way to log out since the menu is gone.

**Review by petertgiles — CHANGES_REQUESTED** [review]
A bit of a corner case but I think it's worth considering.

File: frontend/talentsearch/src/js/components/Router.tsx
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/Router.tsx:601]: Nice catch, hmm I don't have a quick fix for this right now so I'll remove it and make an issue.

**Review by petertgiles — CHANGES_REQUESTED** [review]
I pushed it out to dev and it looks great!  A few straggling issues (some of which may be handled elsewhere?)

- IAP route broken
image
- Exec jobs dead link
image
- Need to merge main to fix lorem
image
- Concept to code link dead
image

**Comment by petertgiles** [comment]
Bonus: It's weird that this link points to login if you're not logged in.  I think it should point to /talent/profile and let routing handle the log in.
image

**Comment by petertgiles** [comment]
> I just merged main, both @petertgiles and @yonikid15 references issues that were resolved in [#4095](https://github.com/gctc-ntgc/gc-digital-talent/issues/4442) which was merged earlier today. Can you both re-view?

Whoah, I think it did fix all my issues!  Nice!

**Review by petertgiles — APPROVED** [review]

**Comment by tristan-orourke** [comment]
Thanks everyone!

**Pull request merged**
